tinyagentos/secrets.py

```
@@ -44,23 +48,96 @@ def _get_key() -> bytes:
+    # Generate a fresh random 32-byte key.
+    raw = os.urandom(32)
+    key_path.parent.mkdir(parents=True, exist_ok=True)
+    tmp = key_path.with_suffix(".tmp")
+    tmp.write_bytes(raw)
+    os.chmod(tmp, 0o600)
+    tmp.rename(key_path)
+    _fernet_key_cache[cache_key] = raw
+    return raw
+
+
+def _fernet_token(key_dir: Path) -> "Fernet":
```

Ruff (0.15.15)

[error] Undefined name `Fernet`

(F821)